tests/run.go

```diff
@@ -205,11 +205,16 @@ func Run(o *option.Option) {
 		})
 
 		ginkgo.It("should share PID namespace with host with --pid=host", func() {
-			command.Run(o, "run", "-d", "--name", testContainerName, "--pid", "host", defaultImage, "sleep", "infinity")
-			processes := command.StdoutStr(o, "top", testContainerName)
-			pid := strings.Fields(processes)[9]
-			output := command.StdOutAsLines(o, "exec", "-d", testContainerName, "ps", "-o", "pid")
-			gomega.Expect(output).Should(gomega.ContainElement(pid))
+			command.Run(o, "run", "-d", "--name", testContainerName, "--pid=host", defaultImage, "sleep", "infinity")
+			pid := command.StdoutStr(o, "inspect", "--format", "{{.State.Pid}}", testContainerName)
+			output := command.StdoutStr(o, "exec", testContainerName, "sh", "-c", fmt.Sprintf("ps ax | grep %s | awk '{print $1}'", pid))
+			gomega.Expect(output).Should(gomega.ContainSubstring(pid))
```

Comment:
Before: There's a process with `pid` running.
After: There's a process with `pid` running, and its command is `sleep`. There could still be a false positive, but I feel that the possibility is low enough. WDYT?

```suggestion
			command.Run(o, "exec", testContainerName, "sh", "-c", fmt.Sprintf("ps -o pid,comm | grep '%s sleep'", pid))
```

Reply:
> It seems that top is replaced with inspect. Could you elaborate more re. why that solves the issue?
This bug is not related to `top` command. I replaced it with `inspect` because `top` command requires cgroup v2 and we might not want to add the restriction for this test spec.

Reply:
TL;DR I think that this PR is an improvement, so we can merge it. However, we're still not sure about the root cause of the bug, so we may not want to close the corresponding issue at least for now. We should keep an eye on the CI after this PR is merged (e.g., maybe monitor a month) in the hope that the bug would somehow go away.

> This bug is not related to top command. I replaced it with inspect because top command requires cgroup v2 and we might not want to add the restriction for this test spec.

💯 

> I'm not sure whether it's because the ps -o pid doesn't list all processes like ps aux
> Or we can also add -a to the ps command to be more secure. WDYT?

It seems that [the `ps` included in the alpine image](https://boxmatrix.info/wiki/Property:ps) does not include any option though:

```sh
➜  ~ lima nerdctl run alpine ps --help
BusyBox v1.35.0 (2022-11-19 10:13:10 UTC) multi-call binary.

Usage: ps [-o COL1,COL2=HEADER] [-T]

Show list of processes

	-o COL1,COL2=HEADER	Select columns for display
	-T			Show threads
```

> or there is a -d flag with the exec command.

Hmm, maybe it's also not the case

```sh
➜  ~ lima nerdctl exec -d sleep ps -o pid
PID
    1
   13
```

Reply:
Revised the PR as the dicussion.